### PR TITLE
WT-13563 Modify dist/prototypes.py to split public and private functions into separate headers

### PR DESCRIPTION
**Pull request by ajmorton**
Update the `prototypes.py` script to create dedicated `*.h` and `*_private.h` header files in each component folder.
Previously all function prototypes were defined in `extern.h`. This script allows modules to define all the functions owned by the module in headers inside that module.

The files also support adding content manually. Only code between the `DO NOT EDIT` flags will be automatically updated. this allows developers to place other items (macros, structs) inside the files.

This PR only introduces the changes to allow creation of these files on a per-module basis. Future tickets will enable it for modules by adding them to the `MODULARISED_COMPONENTS` field.

### CODE REVIEW

**Comment by ajmorton** [comment]
@Siddhartha8899 this is a re-delivery of https://github.com/wiredtiger/wiredtiger/pull/11064. The only change is the last commit https://github.com/wiredtiger/wiredtiger/pull/11083/commits/880a1dfc6c7d14f26d7a44c049149bad774ac089 which updates some more s_all scripts to handle the following cases:
- Headers can now exist in `src/module/` folders
- s_fast should run `prototypes.py` when we modify the script. Without this adding a module to the `SELF_CONTAINED_MODULES` list and running s_fast does nothing.
- Special handing in `function.py` to not skip the entire contents of the new header files. We want to skip the function declarations, but not the definitions of the inline functions

File: dist/prototypes.py [hunk 1]
```diff
@@ -39,7 +39,8 @@
 
 """
 
-if not [f for f in filter_if_fast(source_files(), prefix="../")]:
+files = [f for f in source_files()]
+if not [f for f in filter_if_fast(files + ["../dist/prototypes.py"], prefix="../")]:
```
Comment (Siddhartha8899) [inline dist/prototypes.py:43]: Would it exit earlier, if no file changes were made, and only prototypes.py was changed?

Comment (Siddhartha8899) [reply dist/prototypes.py:43]: If that's the case, doesn't it hold true for all the s_all files?

**Review by Siddhartha8899 — APPROVED** [review]
LGTM mod 1 comment

File: dist/prototypes.py (hunk 1 above)
Comment (ajmorton) [reply dist/prototypes.py:43]: Good point.. I think this needs fixing for all files. I'll spin up a new ticket for this

**Pull request merged**